src/effects/VST3/VST3Effect.h

```diff
@@ -0,0 +1,131 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file VST3Effect.h
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+
+#pragma once
+
+#include <pluginterfaces/gui/iplugview.h>
+#include <pluginterfaces/vst/ivstaudioprocessor.h>
+#include <public.sdk/source/vst/hosting/module.h>
+
+#include "EffectInterface.h"
+
+#include "SampleCount.h"
+#include "effects/Effect.h"
+#include "internal/ComponentHandler.h"
+
+namespace Steinberg
+{
+   namespace Vst
+   {
+      class IComponent;
+      class IEditController;
+      class IParameterChanges;
+      class IComponentHandler;
+   }
+}
+
+class ParameterChangesProvider;
+
+/**
+ * \brief Objects of this class connect Audacity with VST3 effects
+ */
+class VST3Effect final : public EffectUIClientInterface
+{
+   //Keep strong reference to a module while effect is alive
+   std::shared_ptr<VST3::Hosting::Module> mModule;
+
+   const VST3::Hosting::ClassInfo mEffectClassInfo;
+   
+   Steinberg::IPtr<Steinberg::Vst::IComponent> mEffectComponent;
+   Steinberg::IPtr<internal::ComponentHandler> mComponentHandler;
+   internal::ComponentHandler::PendingChangesPtr mPendingChanges;
+   Steinberg::IPtr<Steinberg::Vst::IEditController> mEditController;
+   Steinberg::IPtr<Steinberg::IPlugView> mPlugView;
```

Comment:
Please add comments about which of the above are invariantly non-null, after construction.  I think that includes mEffectComponent, not mComponentHandler; and mPendingChanges is sure to be non-null provided that mComponentHandler is not null.